cli/core/test/e2e/framework/plugin_operations.go

```diff
@@ -0,0 +1,299 @@
+// Copyright 2021 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package framework
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	plugintypes "github.com/vmware-tanzu/tanzu-framework/apis/cli/v1alpha1"
+)
+
+// PluginOps helps to perform plugin operations like create plugin bundle, and publish
+type PluginOps interface {
```

Comment:
We should rename this to something more accurate. One possibility is to break this interface into meaningful and functionality-based interfaces something like below.

```
type GeneratePluginOps interface {
	// GenerateScriptBasedPluginBinaries generates script-based plugin binaries for given plugin metadata and return generated plugin binary file paths
	GenerateScriptBasedPluginBinaries(pluginsMeta []*PluginMeta) ([]string, []error)
}
```
The reason the above function doesn't belong to `PublishPluginOps` is because the implementation of `GenerateScriptBasedPluginBinaries` will remain the same even if we create `localOCIPublishPluginOps` or `localFileBasedPluginOps`. 


```
type PublishPluginOps interface {
	// PublishPluginBinary publishes the plugin binaries to given registry bucket and returns the plugin distribution urls
	PublishPluginBinary(pluginsInfo []*PluginMeta) (distributionUrls []string, errs []error)

	// GeneratePluginBundle generates plugin bundle in local file system for given plugin metadata
	GeneratePluginBundle(pluginsMeta []*PluginMeta) ([]string, []error)

	// PublishPluginBundle publishes the plugin bundles to given registry bucket and returns the plugins discovery urls
	PublishPluginBundle(pluginsInfo []*PluginMeta) (discoveryUrls []string, errs []error)
}
```

```
type PluginLifecycleOps interface {
	// ListPlugins lists all plugins by running the 'tanzu plugin list' command
	ListPlugins() ([]PluginListInfo, error)

	InstallPlugin(...) error
}
```

```
type PluginSourceOps interface {
	// AddPluginDiscoverySource adds plugin discovery source
	AddPluginDiscoverySource(discoveryOpts *DiscoveryOptions) error
}
```


Let me know your thoughts on this.